components/style/values/specified/grid.rs

```diff
@@ -219,28 +210,21 @@ impl Parse for TrackList<LengthOrPercentage> {
                             return Err(StyleParseError::UnspecifiedError.into())
                         }
 
-                        list_type = TrackListType::Auto(values.len() as u16);
+                        list_type = TrackListType::Auto(values.len() as u16 + auto_offset);
                         auto_repeat = Some(repeat);
                         let vec = mem::replace(&mut current_names, vec![]);
                         names.push(vec.into_boxed_slice());
-                        continue
+                        continue;
                     },
                     RepeatType::Fixed => (),
                 }
 
-                // If the repeat count is numeric, we axpand and merge the values.
-                let mut repeat = repeat.expand();
-                let mut repeat_names_iter = repeat.line_names.iter();
-                for (size, repeat_names) in repeat.track_sizes.drain(..).zip(&mut repeat_names_iter) {
-                    current_names.extend_from_slice(&repeat_names);
-                    let vec = mem::replace(&mut current_names, vec![]);
-                    names.push(vec.into_boxed_slice());
-                    values.push(size);
-                }
-
-                if let Some(names) = repeat_names_iter.next() {
-                    current_names.extend_from_slice(&names);
+                let vec = mem::replace(&mut current_names, vec![]);
+                names.push(vec.into_boxed_slice());
+                if let RepeatCount::Number(num) = repeat.count {
+                    auto_offset += (num.value() - 1) as u16;
```

Comment:
This doesn't seem obvious as it sounds. Please drop a comment about this 😄

Reply:
Done. I added the comment to the binding definition.